modules/pam_motd/pam_motd.c

```diff
@@ -166,9 +166,24 @@ static int compare_strings(const void *a, const void *b)
     }
 }
 
+static const char* motddir;
+
 static int filter_dirents(const struct dirent *d)
 {
-    return (d->d_type == DT_REG || d->d_type == DT_LNK);
+    struct stat s;
+    char *fullpath;
+    int rc;
+
+    while(1){
+        if ( d->d_type==DT_DIR ) break;
+        if ( join_dir_strings( &fullpath, motddir, d->d_name )<=0) break;
+        rc=stat( fullpath, &s );
+        free(fullpath);     /* free the memory alloc'ed by join_dir_strings */
+        if ( rc!=0 ) break; /* if the stat() somehow failed */
+
+        return S_IFREG==(s.st_mode&S_IFMT) || S_IFLNK==(s.st_mode&S_IFMT);
```

Comment:
`stat()` will never resturn `S_IFLINK`. To match original condition, it should be `lstat()`.

Reply:
right, I missed that.